libraries/_executor.rb

```diff
@@ -68,6 +68,7 @@ def initialize(options = {})
     #
     def execute!(*pieces)
       command_options = pieces.last.is_a?(Hash) ? pieces.pop : {}
+      cli_user, cli_password = options.fetch(:cli_credentials, [])
```

Comment:
Can we just have two separate options here: `username` and `password`. They should be set via `node.run_state` values in a similar way that the private key is.